testutil/long_running_network.go

```diff
@@ -0,0 +1,287 @@
+package testutil
+
+import (
+	"context"
+	"crypto/rand"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/ava-labs/simplex"
+	"github.com/stretchr/testify/require"
+)
+
+// Set this to a higher value so we don't overload lagging nodes with replication requests
+const longRunningMaxRoundWindow = 100
+
+type LongRunningInMemoryNetwork struct {
+	*InMemNetwork
+
+	stopped atomic.Bool
+}
+
+func NewDefaultLongRunningNetwork(t *testing.T, numNodes int) *LongRunningInMemoryNetwork {
+	nodes := make([]simplex.NodeID, numNodes)
+	for i := range numNodes {
+		nodes[i] = generateNodeID(t)
+	}
+
+	net := NewInMemNetwork(t, nodes)
+	for _, nodeID := range nodes {
+		node := NewSimplexNode(t, nodeID, net, &TestNodeConfig{
+			BlockBuilder:       NewNetworkBlockBuilder(t),
+			ReplicationEnabled: true,
+			MaxRoundWindow:     longRunningMaxRoundWindow,
+		})
+		node.l.SetPanicOnError(true)
+		node.l.SetPanicOnWarn(true)
+	}
+	return &LongRunningInMemoryNetwork{
+		InMemNetwork: net,
+	}
+}
+
+func (n *LongRunningInMemoryNetwork) StartInstances() {
+	require.Equal(n.t, len(n.nodes), len(n.Instances))
+
+	for i := len(n.nodes) - 1; i >= 0; i-- {
+		n.Instances[i].Start()
+	}
+
+	amount := simplex.DefaultEmptyVoteRebroadcastTimeout / 5
+	go n.UpdateTime(100*time.Millisecond, amount)
+}
+
+func (n *LongRunningInMemoryNetwork) UpdateTime(frequency time.Duration, amount time.Duration) {
```

Comment:
can we replace `amount` with something more meaningful like "increment" ? `amount` is a bit generic and also I think it's often used for things that you don't count, like "amount of sugar".